### PR TITLE
t: suppress `rose host-select` warnings

### PR DESCRIPTION
**Pull request by matthewrmshin**
And remove colon file name from source tree, which may cause failure in
some file systems.

@arjclark @benfitzpatrick please review.

### CODE REVIEW

**Comment by arjclark** [comment]
Looks ok to me. Tests fine. Over to @benfitzpatrick

**Comment by matthewrmshin** [comment]
Sorry @arjclark I have got another change set in to ensure that the tests are still good in the absence of site/user configurations. The following command should now work better:

```
CYLC_CONF_PATH= FCM_CONF_PATH= ROSE_CONF_PATH= rose test-battery
```

**Comment by arjclark** [comment]
Just the one comment otherwise fine. Back to @benfitzpatrick.

File: t/rose-suite-run/20-pgrep-conflict.t [hunk 1]
```diff
@@ -50,7 +50,7 @@ FAKE_SUITE_PID=$!
 run_fail rose suite-run -q --no-gcontrol \
     -C $TEST_SOURCE_DIR/$TEST_KEY_BASE --name=$NAME
 disown $FAKE_SUITE_PID  # Don't report 'Terminated...' stuff on kill.
-kill $FAKE_SUITE_PID
+kill $FAKE_SUITE_PID >'/dev/null' || true
```
Comment (benfitzpatrick) [inline t/rose-suite-run/20-pgrep-conflict.t]: Do you mean 2>'/dev/null'?

**Comment by benfitzpatrick** [comment]
One comment, otherwise looks fine.

File: t/rose-suite-run/20-pgrep-conflict.t (hunk 1 above)
Comment (matthewrmshin) [inline t/rose-suite-run/20-pgrep-conflict.t]: This is now fixed.

**Pull request merged**